### ISSUE TITLE
Amiga 1000 issues

### ISSUE THREAD

**Issue by ami-commo-fan** [jbilander/SDBox-v2#4]
I already submitted on Nicolas par-to-spi page, but so far no response, so reporting here as well. 
Does anyone succeeded to have working unit on Amiga 1000?
I have following issue running the adapter on the Amiga 1000.
The symptoms are the sdcard drive appears and disappears constantly for about 20 seconds and after is no longer accessible. All the time CPU usage is at maximum.
When using the same adapter on amiga with standard serial port, all works well.
I was checking the connections and all the signals needed are present on A1000 par. port so I do not understand why it is not working as expected.
Could that be similar driver issue that prevented working on A3000/4000 and was fixed lately?
I tried both 2.0 and 2.1 drivers both behave the same.

**Comment by salocinx** [comment]
The problem on the A4000D was a timing issue (should be resolved in 2.1 release of the driver). The A3000D did not even initialize any MicroSD card without pull-up resistors on the parallel port data lines. Do you have pull-up resistors at the parallel port data lines in your specific SDBox?

Then, someone else reported a different issue with the A1000, some initialization issue. Finally he found out, that the 68010 CPU is incompatible with the current driver. Reverting to the original 68000 CPU resolved the issue. What CPU type you are using (68000 / 68010 or any accelerator) ?

I myself have tested the SDBox on the A1000 and I did not have any troubles with the latest driver revision 2.1, even without the pull-up resistors.

Does your SDBox has a female parallel port connector or do you use any intermediate adapter to connect a standard SDBox to the special A1000 male parallel port?

**Comment by ami-commo-fan** [comment]
Actualy I made the sdbox for friend (with A1000). I built according the par-spi-scheme, so there is no pull-up resistors. I built with the female connector, but tested myself on my amigas (with gender-changer) and it worked. Of course I did not connect the +5V line which is different on both parallel ports. I expect he has a stock machine, but will let him know about the 010 incompatibility.
So I think good to try is the pull-ups. Is it 10k resistor from VCC (+5V) to each 2-9 dataline?

**Comment by salocinx** [comment]
Yes, I pulled-up all the parallel port pins D0-D7, ACK, BUSY, POUT and SEL. But if I remember correctly, this was not needed on the A1000, only the A3000. Even the [SDBox v1](https://github.com/jbilander/sdbox) worked fine on the A1000. Jorgen later added level shifting ICs to the design in order to have a better translation between 3V3/5V for the SPI signals (MOSI/MISO/etc) between the SD card and ATMEGA micro controller (cp. [SDBox v2](https://github.com/jbilander/sdbox-v2)).

However, before you turn your hardware design inside out, I would first test with some other MicroSD card brands. I always made good experiences with SanDisk so far, lots of other brands currently don't work reliably. Maybe Jorgen can tell you what brand he used successfully (it was something other than SanDisk, but I cannot remember).

**Comment by ami-commo-fan** [comment]
Ok, thanks for suggestions. The SDBox you tested ok with A1000 was just wired without the level shifter?

**Comment by salocinx** [comment]
> Ok, thanks for suggestions. The SDBox you tested ok with A1000 was just wired without the level shifter?

As far as I remember, yes. I tested SDBox v1, which is without level shifter, and it worked on my A1000 with a SanDisk (4GB FAT/FAT16 partition).

**Comment by jbilander** [comment]
On SDBox-v1 there is an IC on the SD Card Adapter-board that does the translation. I think it's an 74LVC125A that is fed with 3.3V. It might be that the MISO driven from the MicroSD drop too much to indicate a HIGH on the 74LVC125A or that the voltage level towards the ATMega is too low, however 3.3V should be enough for indicating a HIGH on the ATMega and also on the 74LVC125A. Anyway you have to put a pull-up resistor like SukkoPera did here:
https://github.com/jbilander/sdbox/issues/2#issuecomment-767763618 

More info on the Catalex Micro SD Card Adapter here:
https://circuitstate.com/tutorials/interfacing-catalex-micro-sd-card-module-with-arduino/
https://forum.arduino.cc/index.php?topic=604921.msg4314215#msg4314215

**Comment by Alectronics** [comment]
This v2B mini SDbox never worked reliable, not even with the newest drivers with the multilanguage FAT95 files on them.
After which my SD0: CF drive was gone and disabled.

So more trouble here sadly. Why I placed the old FAT95 back and removed that Mini micro SD Box driver  version. Its useless to try to fix something so unreliable.
